src/collector/dashboard/accountGrowth.ts

```diff
@@ -0,0 +1,70 @@
+import { subDays, startOfToday, endOfDay } from 'date-fns'
+import { getRepository } from 'typeorm'
+
+import { getDateFromDateTime, convertDbTimestampToDate } from './helpers'
+import { AccountTxEntity } from 'orm'
+
+export interface AccountCountInfo {
+  totalAccount: number
+  activeAccount: number
+}
+
+async function getTotalAccount(
+  until: Date
+): Promise<{
+  date: string
+  total_account_count: number
+}> {
+  const queryBuilder = getRepository(AccountTxEntity)
+    .createQueryBuilder()
+    .select('count(distinct account)', 'total_account_count')
+    .where('timestamp <= :until ', { until: endOfDay(until) })
+
+  const result: {
+    total_account_count: number
+  }[] = await queryBuilder.getRawMany()
+
+  return {
+    date: getDateFromDateTime(until),
+    total_account_count: result.length ? result[0].total_account_count : 0
+  }
+}
+
+async function getDailyActiveAccount(daysBefore?: number): Promise<{ date: string; active_account_count: number }[]> {
+  const queryBuilder = getRepository(AccountTxEntity)
+    .createQueryBuilder()
+    .select(convertDbTimestampToDate('timestamp'), 'date')
+    .addSelect('count(distinct account)', 'active_account_count')
+    .groupBy('date')
+    .orderBy('date', 'ASC')
+    .where('timestamp < :today ', { today: startOfToday() })
+
```

Comment:
Can we just rely on raw query for counting unique count? the query time is crucial here. I have experienced that above query was significantly faster than count(distinct account).

```
fcd=> EXPLAIN ANALYZE SELECT COUNT(account) FROM (SELECT DISTINCT account FROM account_tx WHERE timestamp >='2020-04-18' AND timestamp < '2020-05-17') a;

 Aggregate  (cost=1910387.93..1910387.94 rows=1 width=8) (actual time=3906.017..3906.017 rows=1 loops=1)
   ->  HashAggregate  (cost=1909951.07..1910145.23 rows=19416 width=45) (actual time=3798.338..3881.513 rows=416933 loops=1)
         Group Key: account_tx.account
         ->  Bitmap Heap Scan on account_tx  (cost=133587.14..1894239.23 rows=6284739 width=45) (actual time=424.311..1854.403 rows=9014411 loops=1)
               Recheck Cond: (("timestamp" >= '2020-04-18 00:00:00'::timestamp without time zone) AND ("timestamp" < '2020-05-17 00:00:00'::timestamp without time zone))
               Rows Removed by Index Recheck: 31
               Heap Blocks: exact=47284 lossy=164614
               ->  Bitmap Index Scan on account_tx_timestamp  (cost=0.00..132015.96 rows=6284739 width=0) (actual time=409.005..409.005 rows=9016374 loops=1)
                     Index Cond: (("timestamp" >= '2020-04-18 00:00:00'::timestamp without time zone) AND ("timestamp" < '2020-05-17 00:00:00'::timestamp without time zone))
 Planning time: 1.896 ms
 Execution time: 3909.978 ms
```
```
fcd=> EXPLAIN ANALYZE SELECT COUNT(DISTINCT account) FROM account_tx WHERE timestamp >='2020-04-18' AND timestamp < '2020-05-17';

 Aggregate  (cost=1909956.07..1909956.08 rows=1 width=8) (actual time=42255.983..42255.984 rows=1 loops=1)
   ->  Bitmap Heap Scan on account_tx  (cost=133587.14..1894244.23 rows=6284739 width=45) (actual time=413.322..1901.046 rows=9014411 loops=1)
         Recheck Cond: (("timestamp" >= '2020-04-18 00:00:00'::timestamp without time zone) AND ("timestamp" < '2020-05-17 00:00:00'::timestamp without time zone))
         Rows Removed by Index Recheck: 31
         Heap Blocks: exact=47284 lossy=164614
         ->  Bitmap Index Scan on account_tx_timestamp  (cost=0.00..132015.96 rows=6284739 width=0) (actual time=403.426..403.426 rows=9016374 loops=1)
               Index Cond: (("timestamp" >= '2020-04-18 00:00:00'::timestamp without time zone) AND ("timestamp" < '2020-05-17 00:00:00'::timestamp without time zone))
 Planning time: 0.128 ms
 Execution time: 42256.945 ms
````